### PR TITLE
Extract the parseModifier to a public method

### PR DESCRIPTION
**Pull request by vguzmanp**
It can be used with EventKeys and EventButtons, although they don't have any common interface with the 'State' property

#### Things to review:
- [ ] Unit tests
- [ ] No hardcoded resources (strings, sizes, colors...)
- [ ] Sorted usings
- [ ] Add summaries where needed 

If needed, you can add new checkboxes to https://github.com/fluendo/VAS/blob/master/.github/PULL_REQUEST_TEMPLATE.md

### CODE REVIEW

File: VAS.Core/Common/Keyboard.cs [hunk 1]
```diff
@@ -35,37 +36,25 @@ public string NameFromKeyval (uint keyval)
 
 		public HotKey ParseEvent (object obj)
 		{
-			int modifier = 0;
-			EventKey evt = obj as EventKey;
+			EventKey eventKey = obj as EventKey;
 
-			if (IsSupportedModifier (evt.Key)) {
+			int keyval = 0;
```
Comment (ylatuya) [inline VAS.Core/Common/Keyboard.cs]: Why is keyval intialized here now?

File: VAS.Core/Common/Keyboard.cs [hunk 2]
```diff
@@ -74,6 +63,27 @@ public HotKey ParseEvent (object obj)
 			};
 		}
 
+		public int ParseModifier (object originalEvent)
```
Comment (ylatuya) [inline VAS.Core/Common/Keyboard.cs]: Why is this function public but not added to the interface?
 Why can't you use ParseEvent and keep the interface simplier?

File: VAS.Core/Common/Keyboard.cs [hunk 3]
```diff
@@ -187,4 +197,32 @@ bool IsSupportedModifier (Key key)
...
+	/// This class wraps over both EventKey and EventButton and exposes their State property.
+	/// </summary>
+	class EventWrapper
+	{
+		EventKey eventKey;
+		Gdk.EventButton eventButton;
+
+		public EventWrapper (object evnt)
+		{
+			EventKey key = evnt as EventKey;
```
Comment (ylatuya) [inline VAS.Core/Common/Keyboard.cs]: Why do you need these local variables at all ?
eventKey =  evnt as Gdk.EventKey;
eventButton = event as Gdk.EventButton;
if (eventButton != null && eventKey != null) throw new

File: VAS.Core/Common/Keyboard.cs (hunk 1 above)
Comment (vguzmanp) [reply VAS.Core/Common/Keyboard.cs]: Oops

File: VAS.Core/Common/Keyboard.cs (hunk 2 above)
Comment (vguzmanp) [reply VAS.Core/Common/Keyboard.cs]: It exposes a gtk-only way of doing things. Other platforms don't necessarily use an int to store modifier flags, so I thought it wouldn't make sense in the interface.

File: VAS.Core/Common/Keyboard.cs (hunk 3 above)
Comment (vguzmanp) [reply VAS.Core/Common/Keyboard.cs]: Ok

File: VAS.Core/Common/Keyboard.cs (hunk 2 above)
Comment (ylatuya) [reply VAS.Core/Common/Keyboard.cs]: It exposes the modifiers which are not gtk specific, under the HotkKey.Modifier property.
You are creating a new API when you can use an existing one:
int modifier = KeyBoard.ParseEvent(event).Modifier

Comment (vguzmanp) [reply VAS.Core/Common/Keyboard.cs]: Right, I'll move it to the interface.

Calling ParseEvent has the problem that HotKeys have a `Key` property that doesn't make sense here, as it's a mouse press, and I think that Hotkey is good as it is, without having extra logic to manage mouse/keyboard.

Comment (ylatuya) [reply VAS.Core/Common/Keyboard.cs]: ok!

**Pull request merged**